extensions/import/src/wizard/flatFileWizard.ts

```diff
@@ -38,14 +38,25 @@ export class FlatFileWizard {
 
 		let pages: Map<number, ImportPage> = new Map<number, ImportPage>();
 
-		let connectionId = (await azdata.connection.getCurrentConnection())?.connectionId ??
-			(await azdata.connection.openConnectionDialog())?.connectionId;
+		let currentConnection = await azdata.connection.getCurrentConnection();
 
-		if (!connectionId) {
-			vscode.window.showErrorMessage(localize('import.needConnection', "Please connect to a server before using this wizard."));
-			return;
+		let connectionId: string;
+
+		if (!currentConnection) {
+			connectionId = (await azdata.connection.openConnectionDialog(['MSSQL'])).connectionId;
+			if (!connectionId) {
+				vscode.window.showErrorMessage(localize('import.needConnection', "Please connect to a server before using this wizard."));
+				return;
+			}
+		} else {
+			if (currentConnection.providerId !== 'MSSQL') {
+				vscode.window.showErrorMessage(localize('import.needSQLConnection', "The import extension currently support Microsoft SQL databases only. Please connect to a Microsoft SQL database before using this wizard."));
```

Comment:
`currently supports Microsoft SQL databases`

I'd also run this by a PM first to make sure this is the correct product naming to use.

Reply:
Maybe something like `"SQL Server Import extension does not support this type of connection"` is fine.